db/migrate/20211001151212_update_patient_table_string_lengths.rb

```diff
@@ -0,0 +1,191 @@
+    remove_index :patients, name: "patients_index_chain_6"
+    remove_index :patients, name: "patients_index_chain_7"
+    remove_index :patients, name: "patients_index_chain_8"
+    remove_index :patients, name: "patients_index_chain_two_1"
+    remove_index :patients, name: "patients_index_chain_two_2"
+    remove_index :patients, name: "patients_index_chain_three_1"
+  end
+
+  def add_current_indexes
+    add_index :patients, :assigned_user
+    add_index :patients, :creator_id
+    add_index :patients, :date_of_birth
+    add_index :patients, :first_name
+    add_index :patients, :id
+    add_index :patients, :jurisdiction_id
+    add_index :patients, :last_date_of_exposure
+    add_index :patients, :potential_exposure_country
+    add_index :patients, :public_health_action
+    add_index :patients, :purged
+    add_index :patients, :responder_id
+    add_index :patients, :sex
+    add_index :patients, :submission_token
+    add_index :patients, :user_defined_id_cdc
+    add_index :patients, :user_defined_id_nndss
+    add_index :patients, :user_defined_id_statelocal
+    add_index :patients, [:monitoring, :purged, :public_health_action, :isolation, :jurisdiction_id], name: 'patients_index_chain_1'
+    add_index :patients, [:monitoring, :purged, :isolation, :jurisdiction_id], name: 'patients_index_chain_2'
+    add_index :patients, [:last_name, :first_name], name: 'patients_index_chain_3'
+    add_index :patients, [:id, :monitoring, :purged, :isolation, :symptom_onset], name: 'patients_index_chain_4'
+    add_index :patients, [:monitoring, :purged, :isolation, :id, :public_health_action], name: 'patients_index_chain_5'
+    add_index :patients, [:isolation, :jurisdiction_id], name: 'patients_index_chain_6'
+    add_index :patients, [:monitoring, :purged, :isolation], name: 'patients_index_chain_7'
+    add_index :patients, [:jurisdiction_id, :assigned_user], name: 'patients_index_chain_8'
+    add_index :patients, [:primary_telephone, :responder_id, :id, :jurisdiction_id], name: 'patients_index_chain_two_1'
+    add_index :patients, [:email, :responder_id, :id, :jurisdiction_id], name: 'patients_index_chain_two_2'
+    add_index :patients, [:jurisdiction_id, :isolation, :purged, :assigned_user], name: 'patients_index_chain_three_1'
+  end
+
+  # NOTE: removing and adding indexes significantly improves performance
+
+  def up
+    STRING_FIELDS.each do |field|
+      if Patient.where("length(#{field}) > 200").exists?
+        raise StandardError.new "Deteced string value over 200 characters in column #{field}. Aborting migration."
+      end
+    end
+
+    ActiveRecord::Base.record_timestamps = false
+
+    remove_current_indexes
+
+    STRING_FIELDS.each { |field| change_column :patients, field, :string, limit: NEW_LIMIT }
+    BINARY_FIELDS.each { |field| change_column :patients, field, :binary, limit: NEW_LIMIT }
+
+    add_current_indexes
+
+    ActiveRecord::Base.record_timestamps = true
```

Comment:
Actually one more thing (and same for the `down`):
```suggestion
    begin
        STRING_FIELDS.each { |field| change_column :patients, field, :string, limit: NEW_LIMIT }
        BINARY_FIELDS.each { |field| change_column :patients, field, :binary, limit: NEW_LIMIT }
    ensure
        add_current_indexes
        ActiveRecord::Base.record_timestamps = true
    end
```

Reply:
done: https://github.com/SaraAlert/SaraAlert/pull/1107/commits/4b0b811fb2d12f7e301a0457b9eb50977ce7d4a5